flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -692,6 +692,17 @@ public void testDifferentBufferSizeForSubpartitions() throws IOException {
         assertEquals(12, subpartition0.pollBuffer().buffer().getSize());
         assertEquals(5, subpartition1.pollBuffer().buffer().getSize());
         assertEquals(4, subpartition1.pollBuffer().buffer().getSize());
+
+        // when: Set the different buffers size.
+        subpartition0.bufferSize(12);
+        subpartition1.bufferSize(5);
+
+        // and: Add the buffer.
+        bufferWritingResultPartition.broadcastRecord(ByteBuffer.allocate(5));
+
+        // then: The buffer less or equal to configured.
+        assertEquals(5, subpartition0.pollBuffer().buffer().getSize());
+        assertEquals(5, subpartition1.pollBuffer().buffer().getSize());
```

Comment:
Please, take a look more carefully at this test(`testDifferentBufferSizeForSubpartitions`), there are several scenarios that were tested here(send buffer less/greater/equal than the buffer size, change the buffer size and send the again). But you have only one test case which is not enough. So please, increase your test coverage.
Also I suggest extracting your test in the separated method like `testDynamicBufferSizeForBroadcast` or something similar.

Reply:
I add testDynamicBufferSizeForBroadcast for less/greater/equal case